lms/djangoapps/certificates/models.py

```diff
@@ -878,20 +889,56 @@ def is_enabled_for_course(cls, course_key):
         except cls.DoesNotExist:
             return False
         else:
-            return latest.enabled
+            return latest.can_self_generate
```

Comment:
You might need to also check `enabled` as well since there will be some time between when this code is deployed and when the migration is run to populate `can_self_generate`. You could probably just use `or` for this: `return latest.can_self_generate or latest.enabled`.

Reply:
In: https://openedx.atlassian.net/wiki/spaces/OpenDev/pages/23003228/Everything+About+Database+Migrations#EverythingAboutDatabaseMigrations-Makingamigrationtomodifyanexistingtable it says migrations are run in prod before the new code is deployed. Which makes me believe this is safe, is there someone I should check with to verify this?

Reply:
I'm not sure, maybe @andy-armstrong ? I'd be cautious about how much we rely on edx deployment practices though, since this is an open-source repo and other organizations using it may not deploy in the same way we do.